utils/utils.py

```diff
@@ -364,10 +366,69 @@ def check_support(current_params, unsupported_params):
 
 def get_dir(dir_id):
     try:
-        return directories[dir_id]
+        directory = directories[dir_id]
     except KeyError:
         raise ValueError(f"{dir_id} is not known")
+    if not os.path.exists(directory):
+        os.mkdir(directory)
+    return directory
 
 
 def get_ny_taxi_dataset_size(dfiles_num):
     return sum(list(ny_taxi_data_files_sizes_MB.values())[:dfiles_num])
+
+
+def get_tmp_filepath(filename, tmp_dir=None):
+    if tmp_dir is None:
+        tmp_dir = get_dir("tmp")
+
+    filename, extension = os.path.splitext(filename)
+
+    # filename would be transormed like "census-fsi.csv" -> "ROOT_RESOPOSITORY_DIR/tmp/census-fsi-f15cxc9y.csv"
+    file_descriptor, file_path = mkstemp(suffix=extension, prefix=filename + "-", dir=tmp_dir)
+    os.close(file_descriptor)
+
+    return file_path
+
+
+class files_combiner:
```

Comment:
The class name format must be consistent throughout the project.
```suggestion
class FilesCombiner:
```

Reply:
done